### ISSUE TITLE
Helm operator: are rollback workarounds for helm install and upgrade failure still valid

### ISSUE THREAD

**Issue by mikeshng** [operator-framework/operator-sdk#4285]
#### What did you do?
I am looking at the code and I see in the helm operator manager functions we have workarounds for helm install and upgrade error scenarios. 

https://github.com/operator-framework/operator-sdk/blob/v1.2.0/internal/helm/release/manager.go#L172
https://github.com/operator-framework/operator-sdk/blob/v1.2.0/internal/helm/release/manager.go#L213

I am wondering if these are still valid. It seems like there is a new param call `--atomic` that does the samething as the current workaround but not exactly the same. By using the `--atomic` flag the `--wait` flag is automatically triggered as well which is different from the current operator-sdk helm operator behaviour where there is no `--wait`.

I am also wondering how safe is it to perform the uninstall rollback automatically without the user knowing. 

Consider the following scenario: 
1) there is a helm app installed already and went thru a few upgrades.
2) the helm operator sync function incorrectly determines that the helm app is not installed.
https://github.com/operator-framework/operator-sdk/blob/v1.2.0/internal/helm/release/manager.go#L114-L117
https://github.com/operator-framework/operator-sdk/blob/v1.2.0/internal/helm/release/manager.go#L141-L144
if all the helm query results were unable to decode it will return size 0 incorrectly.
https://github.com/helm/helm/blob/976d668dec032f7ce309f31f65a50cebb9b0be95/pkg/storage/driver/secrets.go#L132-L141
3) the helm operator will now proceed with the install path
4) the helm operator might fail the helm install and proceed to perform the helm uninstall automatically.

#### What did you expect to see?
If we can take a second look at the current rollback workaround

#### What did you see instead? Under which circumstances?
in rare scenarios where the kuberenetes api server is misbehaving.

#### Environment

**Operator type:**
/language helm

**Kubernetes cluster type:**

`kind`

`$ operator-sdk version`

`1.2.0`

`$ kubectl version`

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:56:40Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2020-01-14T00:09:19Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by estroz** [comment]
>if all the helm query results were unable to decode it will return size 0 incorrectly.

In what scenarios would a release decode incorrectly, considering that these release objects are generated (afaict)? Is this considered a bug that should be fixed upstream?

As far as being able to use `--wait`-like functionality in helm-operator, check out #4102 as to why it doesn't exist yet. If `--wait` did exist perhaps accepting `--atomic` to get rid of the current workaround would be feasible.

/triage support

**Comment by camilamacedo86** [comment]
HI @jberkhahn, 

As we spoke in the triage meeting, we have the plan to replace the operator helm based code image by the code implementation in : https://github.com/joelanford/helm-operator. More info see the EP in WIP: https://github.com/operator-framework/enhancements/pull/45

The workarounds raised here might no longer be required after that. So, since the goal here shows more remove/change the workaround I'd suggest checking if this need would still be required with the future code first. If not, then we can just close it will be solved hoppfully soon.

**Comment by estroz** [comment]
IIRC we discussed that this feature _might_ not be necessary, but also that the way helm-operator functions is not changing with the rewrite mentioned above. This feature is still worth working on in the meantime, starting with addressing the problems with wait semantics pointed out in #4102.

**Comment by camilamacedo86** [comment]
IMO if we confirm that the workaround is no longer in place in https://github.com/joelanford/helm-operator. then, we can close this one and let the RFE be addressed in #4102.
